file_store/src/file_info_poller.rs

```diff
@@ -0,0 +1,253 @@
+                    for file in files {
+                        if !is_already_processed(&self.db, &cache, &file).await? {
+                            latest_ts = Some(file.timestamp);
+                            cache_file(&cache, &file).await;
+                            send_stream(&sender, &self.store, file).await?;
+                        }
+                    }
+                }
+            }
+        }
+        Ok(())
+    }
+
+    fn after(&self, latest: Option<DateTime<Utc>>) -> DateTime<Utc> {
+        let latest_offset = latest.map(|lt| lt - self.offset);
+        match self.lookback {
+            LookbackBehavior::StartAfter(start_after) => latest_offset.unwrap_or(start_after),
+            LookbackBehavior::Max(max_lookback) => {
+                let max_ts = Utc::now() - max_lookback;
+                latest_offset.map(|lt| lt.max(max_ts)).unwrap_or(max_ts)
+            }
+        }
+    }
+
+    async fn clean(&self, cache: &MemoryFileCache) -> Result {
+        cache.purge(4, 0.25).await;
+        db_clean(&self.db, &self.file_type).await?;
+        Ok(())
+    }
+
+    fn poll_duration(&self) -> std::time::Duration {
+        self.poll_duration.to_std().unwrap_or(DEFAULT_POLL_DURATION)
+    }
+}
+
+async fn send_stream<T>(
+    sender: &Sender<FileInfoStream<T>>,
+    store: &FileStore,
+    file: FileInfo,
+) -> Result
+where
+    T: MsgDecode + TryFrom<T::Msg, Error = Error> + Send + Sync + 'static,
+{
+    let stream = store
+        .stream_file(file.clone())
+        .await?
+        .filter_map(|msg| async {
+            msg.map_err(|err| {
+                tracing::error!(
+                    "Error streaming entry in file of type {}: {err:?}",
+                    std::any::type_name::<T>()
+                );
+                err
+            })
+            .ok()
+        })
+        .filter_map(|msg| async {
+            <T as MsgDecode>::decode(msg)
+                .map_err(|err| {
+                    tracing::error!(
+                        "Error in decoding message of type {}: {err:?}",
+                        std::any::type_name::<T>()
+                    );
+                    err
+                })
+                .ok()
+        })
+        .boxed();
+
+    let incoming_data_stream = FileInfoStream {
+        file_info: file,
+        stream,
+    };
+
+    let _ = sender.send(incoming_data_stream).await;
+    Ok(())
+}
+
+fn create_cache() -> MemoryFileCache {
+    Cache::new()
+}
+
+async fn is_already_processed(
+    db: impl sqlx::PgExecutor<'_>,
+    cache: &MemoryFileCache,
+    file_info: &FileInfo,
+) -> Result<bool> {
+    if cache.get(&file_info.key).await.is_some() {
+        Ok(true)
+    } else {
+        db_exists(db, file_info).await
+    }
+}
+
+async fn cache_file(cache: &MemoryFileCache, file_info: &FileInfo) {
+    cache.insert(file_info.key.clone(), true, CACHE_TTL).await;
+}
+
+async fn db_latest_ts(
```

Comment:
instead of prefixing these with `db_`, consider putting these in an inline module:

```rust
mod db {
   async fn db_latest_ts(...) { ... }
}
```

Reply:
Good idea, fixed c7ea52a